devicetypes/plaidsystems/spruce-controller.src/spruce-controller.groovy

```diff
@@ -14,736 +10,570 @@
+		}
+		else {
+			scheduleTimes.push(zigbee.writeAttribute(ON_OFF_CLUSTER, OFF_WAIT_TIME_ATTRIBUTE, DataType.UINT16, 0, [destEndpoint: endpoint]))
+		}
+	}
+	if (!startRun || totalTime == 0) return noSchedule()
+
+	//start after scheduleTimes are sent
+	scheduleTimes.push(zigbee.command(ON_OFF_CLUSTER, 1, "", [destEndpoint: 1]))
+	sendEvent(name: "status", value: "Scheduled for ${totalTime}min(s)", descriptionText: "Start schedule ending in ${totalTime} mins")
+	return scheduleTimes
 }
-def z11off() {	
-    "st cmd 0x${device.deviceNetworkId} 12 6 0 {}"    
+
+//write switch time settings map
+def settingsMap(WriteTimes, attrType){
+
+	def runTime
+	def sendCmds = []
+	for (endpoint in 1..17) {
+		if (WriteTimes."${endpoint}"){
+			runTime = Integer.parseInt(WriteTimes."${endpoint}")
+
+			if (attrType == ON_TIME_ATTRIBUTE) sendCmds.push(zigbee.writeAttribute(ON_OFF_CLUSTER, ON_TIME_ATTRIBUTE, DataType.UINT16, runTime, [destEndpoint: endpoint]))
+			else sendCmds.push(zigbee.writeAttribute(ON_OFF_CLUSTER, OFF_WAIT_TIME_ATTRIBUTE, DataType.UINT16, runTime, [destEndpoint: endpoint]))
+		}
+	}
+	return sendCmds
 }
-def z12on() {	
-    return manual() + "st cmd 0x${device.deviceNetworkId} 13 6 1 {}"    
+
+//send switch time
+def writeType(endpoint, cycle){
+	zigbee.writeAttribute(ON_OFF_CLUSTER, ON_TIME_ATTRIBUTE, DataType.UINT16, cycle, [destEndpoint: endpoint])
 }
-def z12off() {    
-    "st cmd 0x${device.deviceNetworkId} 13 6 0 {}"    
+
+//send switch off time
+def writeTime(endpoint, runTime){
+	zigbee.writeAttribute(ON_OFF_CLUSTER, OFF_WAIT_TIME_ATTRIBUTE, DataType.UINT16, runTime, [destEndpoint: endpoint])
 }
-def z13on() {	
-	return manual() + "st cmd 0x${device.deviceNetworkId} 14 6 1 {}"   
+
+//set reporting and binding
+def configure() {
+	// Device-Watch allows 2 check-in misses from device, checks every 2 hours
+	sendEvent(name: "DeviceWatch-DeviceStatus", value: "online")
+	sendEvent(name: "healthStatus", value: "online")
+	sendEvent(name: "DeviceWatch-Enroll", value: 2* 60 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+
+	configureHealthCheck()
+
+	String zigbeeId = swapEndianHex(device.hub.zigbeeId)
+	log.debug "Configuring Reporting and Bindings ${device.name} ${device.deviceNetworkId} ${device.zigbeeId}"
+
+	//setup binding for 18 endpoints
+	def bindCmds = []
+	bindCmds += zigbee.addBinding(ON_OFF_CLUSTER, [destEndpoint: 1])
+	bindCmds += zigbee.addBinding(ALARMS_CLUSTER, [destEndpoint: 1])
+
+	for (endpoint in 1..18){
+		bindCmds += zigbee.addBinding(BINARY_INPUT_CLUSTER, [destEndpoint: endpoint])
+	}
+
+	//setup reporting for 18 endpoints
+	def reportingCmds = []
+	reportingCmds += zigbee.configureReporting(ON_OFF_CLUSTER, 0, DataType.BOOLEAN, 1, 0, 0x01, [destEndpoint: 1])
+	reportingCmds += zigbee.configureReporting(ALARMS_CLUSTER, 0, DataType.UINT16, 1, 0, 0x00, [destEndpoint: 1])
```

Comment:
`zigbee.configureReporting` includes the binding command as well so the `addBinding` commands above are redundant.

Reply:
@natec007 Please, could You change the code according to Steven's suggestions ?